runtime/runtime/src/actions.rs

```diff
@@ -394,11 +394,24 @@ pub(crate) fn action_create_account(
     account_id: &AccountId,
     predecessor_id: &AccountId,
     result: &mut ActionResult,
+    current_protocol_version: ProtocolVersion,
 ) {
     if account_id.is_top_level() {
-        if account_id.len() < account_creation_config.min_allowed_top_level_account_length as usize
-            && predecessor_id != &account_creation_config.registrar_account_id
-        {
+        let old_condition_for_error = account_id.len()
+            < account_creation_config.min_allowed_top_level_account_length as usize
+            && predecessor_id != &account_creation_config.registrar_account_id;
+        let new_condition_for_error = !account_id.is_implicit()
+            && predecessor_id != &account_creation_config.registrar_account_id;
+        let is_error = if checked_feature!(
+            "protocol_feature_restrict_tla",
+            RestrictTLA,
+            current_protocol_version
+        ) {
+            new_condition_for_error
+        } else {
+            old_condition_for_error
+        };
```

Comment:
What happens to the now obsolete parameter `min_allowed_top_level_account_length: 32`?
Have you considered to instead just set `min_allowed_top_level_account_length: 65` instead of this code change? Something like `139.yaml`:
```yaml
# Implements NEP-492, disallowing all top-level accounts.
min_allowed_top_level_account_length: { old: 32, new: 65 }
```

I think doing it this way would be less confusing for future development, and it also gives private shards the freedom to overwrite this setting for their purposes.

Reply:
cc @chefsale @mhalambek

Please let us know if you have an opinion on top-level account creation. If Calimero relies on top-level account creation by non-registrar accounts, it would be crucial to know before we stabilize this.

Reply:
> What happens to the now obsolete parameter `min_allowed_top_level_account_length: 32`? Have you considered to instead just set `min_allowed_top_level_account_length: 65` instead of this code change? Something like `139.yaml`:
> 
> ```yaml
> # Implements NEP-492, disallowing all top-level accounts.
> min_allowed_top_level_account_length: { old: 32, new: 65 }
> ```
> 
> I think doing it this way would be less confusing for future development, and it also gives private shards the freedom to overwrite this setting for their purposes.

yeah this is a very good point. I should stop writing code :)